benches/criterion_stark.rs

```diff
@@ -0,0 +1,55 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion, Throughput};
+
+mod functions;
+mod util;
+
+pub fn proof_benchmarks(c: &mut Criterion) {
+    let mut group = c.benchmark_group("STARK");
+    group.sample_size(10);
+
+    for trace_length in [8, 128, 256, 512, 1024].iter() {
+        group.throughput(Throughput::Bytes(*trace_length as u64));
+        group.bench_with_input(
+            BenchmarkId::new("Simple Fibonacci", trace_length),
+            trace_length,
+            |b, &t| {
+                b.iter(|| functions::stark::prove_fib(t));
+            },
+        );
+    }
+
+    group.bench_function("2 column Fibonacci", |bench| {
+        bench.iter(functions::stark::prove_fib_2_cols);
```

Comment:
I'm afraid these may be optimized away by the compiler. I think we should add some kind of input/output and use `black_box` to avoid that.

Reply:
I think marking the functions with `#[inline(never)]` is likely to suffice, and it's necessary anyway for IAI.